lib/crypto/ed25519/ed25519.go

```diff
@@ -71,6 +71,9 @@ func (b PublicKeyBytes) Encode() ([]byte, error) {
 // Decode returns the SCALE decoded PublicKeyBytes
 func (b PublicKeyBytes) Decode(r io.Reader) ([PublicKeyLength]byte, error) {
 	_, err := r.Read(b[:])
+	if errors.Is(err, io.EOF) {
+		return b, nil
+	}
```

Comment:
this seems wrong - if there's an EOF it's because there weren't enough bytes in the reader to create the public key, so there should be an error or else this will create an invalid public key